src/components/partials/user-editor/user-editor.component.js

```diff
@@ -343,88 +352,143 @@ class UserEditor extends Component {
-                          <em>{t("ADMINS.GROUP")}</em>
-                        </MenuItem>
-                        <MenuItem value="field">
-                          <em>{t("ADMINS.FIELD")}</em>
-                        </MenuItem>
-                      </Select>
-                    </FormControl>
-                    <div className="error-messages">{t(errors.adminType)}</div>
-                    <FormControl className="form-input">
-                      <InputLabel id="agency-label">
-                        {t("TABLE.AGENCY")}
-                      </InputLabel>
-                      <Select
-                        readOnly={!!showingOptions.readOnly}
-                        labelId="agency-label"
-                        onChange={(e) =>
-                          setFieldValue("agency", e.target.value)
-                        }
+                    {
+                      isAgencyOrGlobal ? (
+                        <React.Fragment>
+                          <FormControl className="form-input">
+                            <InputLabel id="role-label">
+                              {t("CREATE_USER_PAGE.ROLE")}
+                            </InputLabel>
+                            <Select
+                              readOnly={!!showingOptions.readOnly}
+                              labelId="role-label"
+                              onChange={(e) => {
+                                this.setState({ disabled: isValid });
+                                setFieldValue("adminType", e.target.value);
+                              }}
+                              value={values.adminType}
+                            >
+                              {authService.userRole === "global" ? (
+                                <MenuItem value="global">
+                                  <em>{t("ADMINS.GLOBAL")}</em>
+                                </MenuItem>
+                              ) : (
+                                ""
+                              )}
+                              {authService.userRole === "global" ||
+                              authService.userRole === "agency" ? (
+                                <MenuItem value="agency">
+                                  <em>{t("ADMINS.AGENCY")}</em>
+                                </MenuItem>
+                              ) : (
+                                ""
+                              )}
+                              <MenuItem value="group">
+                                <em>{t("ADMINS.GROUP")}</em>
+                              </MenuItem>
+                              <MenuItem value="field">
+                                <em>{t("ADMINS.FIELD")}</em>
+                              </MenuItem>
+                            </Select>
+                          </FormControl>
+                          <div className="error-messages">{t(errors.adminType)}</div>
+                        </React.Fragment>
+                      ) :(
```

Comment:
...otherwise if not Agency Admin, Global Admin, show the TextField
Are we expecting other values here other than Field Officer?

This way if it's something else, it would show the read-only, which I guess makes sense, since it's an invalid Role anyways